gcovr/configuration.py

```diff
@@ -50,6 +50,30 @@ def check_inputfile(value):
     return os.path.abspath(value)
 
 
+def check_output_file(value):
+    r"""
+    Check if the output file can be created.
+    """
+
+    if value is not None:
+        if os.path.isfile(value) and not os.access(value, os.W_OK):
+            raise ArgumentTypeError(
+                "File {value} exist but is not writable".format(value=value))
+
+        dirname = os.path.dirname(value)
+        if dirname == '':
+            dirname = '.'
+        if not os.path.isdir(dirname):
+            raise ArgumentTypeError(
+                "Directory {dirname} does not exist".format(dirname=dirname))
+
+        if not os.access(dirname, os.W_OK):
+            raise ArgumentTypeError(
+                "Directory {dirname} is not writable".format(dirname=dirname))
```

Comment:
Error messages can be designed along different dimensions:

What the error message tells us:

* stating an error, negative: The directory is not writable.
* positive, describing an expectation or pointing towards a solution: The directory has to be writable.

How the error message is phrased:

* natural phrasing, easier to read: The directory foo/bar has to be writable.
* structured, easier for online searches: The directory has to be writable: foo/bar

I have a mild preference for positive+structured, but won't insist. This is really not important compared to other stuff.

Reply:
I agree with you and will change this.

Reply:
After changing to the python way we have no cover problem anymore and only one error message containing the OS error message:
```python
raise ArgumentTypeError("Could not create output file {value!r}: {error}".format(value=value, error=e.strerror))
```